src/main/java/de/tum/in/www1/artemis/service/connectors/bitbucket/BambooBuildPlanUpdateService.java

```diff
@@ -0,0 +1,184 @@
+        parameters.add("save", "Save repository");
+        parameters.add("bamboo.successReturnMode", "json");
+        parameters.add("repository.git.branch", branchName);
+        parameters.add("repository.git.repositoryUrl", newRepoUrl);
+        parameters.add("repository.git.authenticationType", "PASSWORD");
+        parameters.add("repository.git.passwordCredentialsSource", "SHARED_CREDENTIALS");
+        parameters.add("repository.git.useShallowClones", "true");
+        parameters.add("repository.git.commandTimeout", "180");
+
+        try {
+            String requestUrl = bambooServerUrl + "/chain/admin/config/updateRepository.action";
+            UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(requestUrl).queryParams(parameters);
+            var response = bambooRestTemplate.exchange(builder.build().toUri(), HttpMethod.POST, null, Void.class);
+            log.info("Response from Bamboo updateRepository.action {}", response);
+        }
+        catch (Exception ex) {
+            // TODO: improve error handling
+            String message = "Request failed on the server with response code 500. Make sure all required fields have been provided using the various field and value parameters. "
+                    + "The server log may provide insight into missing fields: " + ex.getMessage();
+            throw new BambooException(message);
+        }
+    }
+
+    /**
+     * Fetches the list of repositories in the given build plan, then tries to find the repository in the bamboo build plan, first using the firstRepositoryName,
+     * second using the secondRepositoryName
+     *
+     * @param firstRepositoryName  the repository name that should be found first
+     * @param secondRepositoryName in case firstRepositoryName is not found, the repository name that should be found second
+     * @param buildPlanKey         the bamboo build plan key in which the repository with the specified name should be found
+     * @return the found bamboo repository in the build plan or null in case the repository with the specified name could not be found
+     */
+    private BambooRepositoryDTO findBambooRepository(String firstRepositoryName, @Nullable String secondRepositoryName, String buildPlanKey) {
+        List<BambooRepositoryDTO> list = this.getBuildPlanRepositoryList(buildPlanKey);
+        var bambooRepository = this.lookupRepository(firstRepositoryName, list);
+        if (bambooRepository == null && secondRepositoryName != null) {
+            bambooRepository = this.lookupRepository(secondRepositoryName, list);
+        }
+        return bambooRepository;
+    }
+
+    private BambooRepositoryDTO lookupRepository(String name, List<BambooRepositoryDTO> list) {
+        var repository = list.stream().filter(repo -> name.equals(repo.getName())).findFirst();
+        if (repository.isPresent()) {
+            return repository.get();
+        }
+
+        if (StringUtils.isNumeric(name)) {
```

Comment:
Add a comment about what "plan B" is here: Act as if the name is an ID.

(Btw, when can the name actually be an ID? 🤔)

Reply:
Not new code, just moved